cibuildwheel/util.py

```diff
@@ -76,40 +77,39 @@ def download(url: str, dest: str) -> None:
         break
 
     try:
-        with open(dest, 'wb') as file:
-            file.write(response.read())
+        dest.write_bytes(response.read())
     finally:
         response.close()
 
 
 class DependencyConstraints:
-    def __init__(self, base_file_path: str):
-        assert os.path.exists(base_file_path)
-        self.base_file_path = os.path.abspath(base_file_path)
+    def __init__(self, base_file_path: Path):
+        assert base_file_path.exists()
+        self.base_file_path = base_file_path.resolve()
 
     @staticmethod
     def with_defaults() -> 'DependencyConstraints':
         return DependencyConstraints(
-            base_file_path=os.path.join(os.path.dirname(__file__), 'resources', 'constraints.txt')
+            base_file_path=Path(__file__).parent / 'resources' / 'constraints.txt'
```

Comment:
`resources_dir` could be nice here

Reply:
Nice! I missed the fact it already existed :-)